cleanlab/outlier.py

```diff
@@ -0,0 +1,534 @@
+       *  method : {"entropy", "least_confidence"}, default="entropy"
+             OOD scoring method.
+
+             Letting ``P = pred_probs[i]`` denote the given predicted class-probabilities
+             for datapoint *i*, its score can either be:
+
+             - ``'entropy'``: ``- sum_{j} P[j] * log(P[j])``
+             - ``'least_confidence'``: ``1 - max(P)``
+
+    """
+
+    OUTLIER_PARAMS = {"k", "t", "knn"}
+    OOD_PARAMS = {"confident_thresholds", "adjust_pred_probs", "method"}
+    DEFAULT_PARAM_DICT: Dict[Union[str, int, None], Union[str, int, None, np.ndarray]] = {
+        "k": None,  # ood features param
+        "t": 1,  # ood features param
+        "knn": None,  # ood features param
+        "adjust_pred_probs": True,  # ood pred_probs param
+        "method": "entropy",  # ood pred_probs param
+        "confident_thresholds": None,  # ood pred_probs param
+    }
+
+    def __init__(self, params: dict = {}):
+        self._assert_valid_params(params, self.DEFAULT_PARAM_DICT)
+        self.params = self.DEFAULT_PARAM_DICT
+        if params is not None:
+            self.params = {**self.params, **params}
+
+    def fit_score(
+        self,
+        *,
+        features: Optional[np.ndarray] = None,
+        pred_probs: Optional[np.ndarray] = None,
+        labels: Optional[np.ndarray] = None,
+        verbose: bool = True,
+    ) -> np.ndarray:
+        """
+        Fits an estimator and returns out-of-distribution scores following optional parameters during object initialization.
+        Scores lie in [0,1] with smaller values indicating examples that are less typical under the dataset
+        distribution (values near 0 indicate outliers). Exactly one of `features` or `pred_probs` needs to be passed
+        in to calculate scores.
+
+        If `features` are passed in a `NearestNeighbors` object is fit. If `pred_probs` and 'labels' are passed in a
+        `confident_thresholds` np.ndarray is fit. For details see :py:func:`fit
+        <cleanlab.outlier.OutOfDistribution.fit>` function.
+
+        Parameters
+        ----------
+        features : np.ndarray, optional
+          Feature array of shape ``(N, M)``, where N is the number of examples and M is the number of features used to represent each example.
+          For details, `features` in the same format expected by the :py:func:`fit <cleanlab.outlier.OutOfDistribution.fit>` function.
+
+        pred_probs : np.ndarray, optional
+          An array of shape ``(N, K)`` of model-predicted probabilities.
+          For details, `pred_probs` in the same format expected by the :py:func:`fit <cleanlab.outlier.OutOfDistribution.fit>` function.
+
+        labels : np.ndarray, optional
+          A discrete vector of noisy labels, i.e. some labels may be erroneous of shape ``(N,)``.
+          For details, `labels` in the same format expected by the :py:func:`fit <cleanlab.outlier.OutOfDistribution.fit>` function.
```

Comment:
"noisy labels, i.e. some labels may be erroneous" -> "given labels for the data"